packages/mds-transaction-api/handlers/get-transactions.ts

```diff
@@ -118,10 +118,23 @@ export const GetTransactionsHandler = async (
   next: express.NextFunction
 ) => {
   try {
+    const { scopes } = res.locals
+
     const order = getOrderOption(req)
-    const { provider_id, before, after } = parseRequest(req)
+    const { provider_id: quereied_provider_id, before, after } = parseRequest(req)
       .single({ parser: String })
       .query('provider_id', 'before', 'after')
+
+    const provider_id = scopes.includes('transactions:read') ? quereied_provider_id : res.locals.claims?.provider_id
+
+    if (provider_id === null) {
+      /* This should never happen -- a client with just the transactions:read:provider scope
+       * should *always* have a provider_id claim in their token.
+       */
+
+      return res.status(500).send({ error: 'internal_server_error' })
+    }
+
```

Comment:
In at least one other instance we have used the `non-null assertion` operator in this scenario when other middleware has previously asserted something exists.

Reply:
Yeah saw both this and the `non-null assertion` being used, so took a gamble. Will happily change to `non-null assertion`.